### ISSUE TITLE
Possible additional question types

### ISSUE THREAD

**Issue by rixx** [pretalx/pretalx#1833]
Here are some question types that we a) could support directly, or b) make sure there's an interface for plugins to support these question types.

|Question type | Validation | Display |
|--|--|--|
| Social media (X, LinkedIn, Fedi, YouTube, Instagram | Valid URL, correct domain (except fedi) | Favicon + username |
| Choice + custom "other" answer | - | - |
| Country | List of countries | - |
| Phone number | - | - |

**Comment by johanvdw** [comment]
I'm not sure questions are always the right way for this: this would make it dependant on the actual conference. Apart from that it also makes it impossible to store data for people who are team members but not speakers.

Maybe our usecase is peculiar, but at FOSDEM we use matrix for chat and would like to store the matrix_id also for people who are team members if they are not a speaker, because they will also get special right in the chat.

**Comment by rixx** [comment]
I’m not sure how additional question types would make it impossible to store data for team members – the additional question types would not change our data structure, and would just add more options (display in forms + display in read-only/public mode) to the existing questions feature.

**Comment by johanvdw** [comment]
I wonder if it would not be better to link some of these fields directly to the user, rather than to Speaker profile.

**Comment by rixx** [comment]
Are we talking about the same thing? Because a) I don't see how adding more types of questions (literally just expanding the enum `Question.variant` + input field handling) changes our data modeling. And b) the object/table storing the answers to questions have nothing to do with `SpeakerProfile`s, they are connected directly to `User` objects. What am I missing?

**Comment by johanvdw** [comment]
Okay, then that probably means that my interpretation was incorrect, sorry for the noise.

**Comment by johanvdw** [comment]
> To improve plugin capabilities here, I think it might be enough to add a new field like `question_identifier` that could be a stable identifier (even across events), and maybe one or two plugin hooks to assist with question creation and updates?

yes please. Especially since the question itself uses localization matching against that one is a pita.
